From a8b73cb281de09f4f6e5b1cef0d93b654c68614b Mon Sep 17 00:00:00 2001
From: Xiao-Long Chen <chenxiaolong@cxl.epac.to>
Date: Tue, 11 Dec 2012 15:31:39 -0500
Subject: [PATCH] Use Fedora-specific library paths

---
 primusrun | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/primusrun b/primusrun
index 81e2dc0..81f40f0 100755
--- a/primusrun
+++ b/primusrun
@@ -26,7 +26,7 @@
 # export PRIMUS_libGLd=${PRIMUS_libGLd:-'/usr/$LIB/libGL.so.1'}
 
 # Directory containing primus libGL
-PRIMUS_libGL=${PRIMUS_libGL:-$(dirname `readlink -ne $0`)/'$LIB'}
+PRIMUS_libGL=${PRIMUS_libGL:-/usr/'$LIB'/primus}
 
 # On some distributions, e.g. on Ubuntu, libnvidia-tls.so is not available
 # in default search paths.  Add its path manually after the primus library
-- 
2.14.3

