From 0e378a469e25e6fe1cb4434ebd836c98cb3297b5 Mon Sep 17 00:00:00 2001
From: Gary Gatling <gsgatlin@ncsu.edu>
Date: Thu, 2 Mar 2017 14:50:47 -0500
Subject: [PATCH] primus needs this variable workaround for libglvnd enabled
 mesa

---
 primusrun | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/primusrun b/primusrun
index 81f40f0..d53572d 100755
--- a/primusrun
+++ b/primusrun
@@ -1,5 +1,8 @@
 #!/bin/bash
 
+# primus needs this variable workaround for libglvnd enabled mesa
+export __GLVND_DISALLOW_PATCHING=1
+
 # Readback-display synchronization method
 # 0: no sync, 1: D lags behind one frame, 2: fully synced
 # export PRIMUS_SYNC=${PRIMUS_SYNC:-0}
-- 
2.14.3

