From ddd757edefdf0a1a1db2a0064abf2394a53e3dfd Mon Sep 17 00:00:00 2001
From: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
Date: Mon, 25 Dec 2017 17:47:07 -0500
Subject: [PATCH 3/3] bumblebee-bugreport: Add support for Fedora

Signed-off-by: Andrew Gunnerson <chenxiaolong@cxl.epac.to>
---
 scripts/bumblebee-bugreport.in | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/scripts/bumblebee-bugreport.in b/scripts/bumblebee-bugreport.in
index efab9e3..e821925 100644
--- a/scripts/bumblebee-bugreport.in
+++ b/scripts/bumblebee-bugreport.in
@@ -40,6 +40,7 @@ PKG_QUERY_TOOLS=(
     'dpkg-query --show'
     'pacman -Q'
     'mpkg-show'
+    'rpm -q'
 )
 PKG_QUERY_TOOL=
 PACKAGES=
@@ -71,6 +72,10 @@ case "$PKG_QUERY_TOOL" in
     PACKAGES=(bumblebeed xf86-video-nouveau mesa libgl nvidia-driver-optimus nvidia-driver-optimus32 virtualgl virtualgl32)
     LIST_DIRS_RECURSIVE=(/usr/lib64/nvidia-optimus /usr/lib/nvidia-optimus)
     ;;
+  rpm*)
+    PACKAGES=(bumblebee nvidia-driver akmod-nvidia xorg-x11-drv-nvidia mesa-libGL VirtualGL primus)
+    LIST_DIRS_RECURSIVE=(/usr/lib64/nvidia /usr/lib/nvidia)
+    ;;
   *)
     # unknown package management tool :/
     ;;
@@ -254,6 +259,9 @@ modinfo nvidia-current
 modinfo nouveau
 
 header "Driver information"
+ls -ld /usr/lib64/libnvidia*
+ls -ld /usr/lib64/libGL.so*
+ls -ld /usr/lib64/libvdpau.so*
 ls -ld /usr/lib/libnvidia*
 ls -ld /usr/lib/libGL.so*
 ls -ld /usr/lib/libvdpau.so*
-- 
2.14.3

